plugins/crypto/main/src/azkaban/crypto/Crypto.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package azkaban.crypto;
+
+import java.util.Base64;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * Crypto class that actually delegates to version specific implementation of ICrypto interface. In other words,
+ * it's a factory method class that implements ICrypto interface
+ */
+public class Crypto implements ICrypto {
```

Comment:
Just a side note regarding the design here,  I feel the Crypto class actually works like a factory class which is responsible for getting the cryptProvider by version and may work better not to implement the ICrypto interface.  the encrypt method in the Crypt Interface therefore doesn't need to know the version as the factory class will be responsible of figuring out the right version, once it is called it just do the work.